src/components/Calculator.jsx

```diff
@@ -1,65 +1,71 @@
-      <div className="display">0</div>
-      <button type="submit" className="calc-key">
+      <div className="display">{result.next || result.total || 0}</div>
+      <button type="submit" className="calc-key" onClick={handleClick}>
         AC
       </button>
-      <button type="submit" className="calc-key">
+      <button type="submit" className="calc-key" onClick={handleClick}>
         +/-
       </button>
-      <button type="submit" className="calc-key">
+      <button type="submit" className="calc-key" onClick={handleClick}>
         %
       </button>
-      <button type="submit" className="operator">
+      <button type="submit" className="operator" onClick={handleClick}>
         ÷
       </button>
-      <button type="submit" className="calc-key">
+      <button type="submit" className="calc-key" onClick={handleClick}>
         7
       </button>
-      <button type="submit" className="calc-key">
+      <button type="submit" className="calc-key" onClick={handleClick}>
         8
       </button>
-      <button type="submit" className="calc-key">
+      <button type="submit" className="calc-key" onClick={handleClick}>
         9
       </button>
-      <button type="submit" className="operator">
+      <button type="submit" className="operator" onClick={handleClick}>
         x
       </button>
-      <button type="submit" className="calc-key">
+      <button type="submit" className="calc-key" onClick={handleClick}>
         4
       </button>
-      <button type="submit" className="calc-key">
+      <button type="submit" className="calc-key" onClick={handleClick}>
         5
       </button>
-      <button type="submit" className="calc-key">
+      <button type="submit" className="calc-key" onClick={handleClick}>
         6
       </button>
-      <button type="submit" className="operator">
+      <button type="submit" className="operator" onClick={handleClick}>
         -
       </button>
-      <button type="submit" className="calc-key">
+      <button type="submit" className="calc-key" onClick={handleClick}>
         1
       </button>
-      <button type="submit" className="calc-key">
+      <button type="submit" className="calc-key" onClick={handleClick}>
         2
       </button>
-      <button type="submit" className="calc-key">
+      <button type="submit" className="calc-key" onClick={handleClick}>
         3
       </button>
-      <button type="submit" className="operator">
+      <button type="submit" className="operator" onClick={handleClick}>
         +
       </button>
-      <button type="submit" className="zero">
+      <button type="submit" className="zero" onClick={handleClick}>
         0
       </button>
-      <button type="submit" className="calc-key">
+      <button type="submit" className="calc-key" onClick={handleClick}>
         .
       </button>
-      <button type="submit" className="operator">
+      <button type="submit" className="operator" onClick={handleClick}>
```

Comment:
- [ ] Based on the requirements, please ensure you use props in your project, you can achieve this by having a child component and passing props  to the child component 👍🏿   It would be better to create a Button component and pass properties like type, value, and onClick 👍🏿